magellan-journey-sample/src/main/java/com/wealthfront/magellan/sample/MainActivity.kt

```diff
@@ -1,21 +1,61 @@
 package com.wealthfront.magellan.sample
 
 import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
 import androidx.appcompat.app.AppCompatActivity
 import com.wealthfront.magellan.lifecycle.setContentScreen
+import com.wealthfront.magellan.sample.App.Provider.appComponent
+import com.wealthfront.magellan.sample.menu.MenuProvider
+import javax.inject.Inject
 
 class MainActivity : AppCompatActivity() {
 
-  private val expeditionProvider = ExpeditionProvider
+  @Inject lateinit var expedition: Expedition
+  @Inject lateinit var menuProvider: MenuProvider
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    setContentScreen(expeditionProvider.expedition)
+    appComponent.inject(this)
+    setContentScreen(expedition)
+    initActionBar()
+  }
+
+  private fun initActionBar() {
+    val actionBar = super.getSupportActionBar()!!
+    actionBar.setDisplayHomeAsUpEnabled(true)
+    actionBar.setDisplayShowTitleEnabled(true)
+  }
+
+  override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+    menuInflater.inflate(R.menu.menu, menu)
+    menuProvider.setMenu(menu)
+    return super.onCreateOptionsMenu(menu)
+  }
+
+  override fun onPrepareOptionsMenu(menu: Menu?): Boolean {
+    menuProvider.setMenu(menu)
+    return super.onPrepareOptionsMenu(menu)
   }
 
   override fun onBackPressed() {
-    if (!expeditionProvider.expedition.backPressed()) {
+    if (!expedition.backPressed()) {
       super.onBackPressed()
     }
   }
+
+  override fun onDestroy() {
+    menuProvider.clearMenu()
```

Comment:
Do we want to make menuProvider a `LifecycleAware` component and do this clearing in `onHide`?

Reply:
Done